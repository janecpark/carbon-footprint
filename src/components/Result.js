import React, { useContext, useState, useEffect } from 'react';
import ResultContext from '../ResultContext';
import Alert from '../Alert';

const Result = () => {
  const { result, setResult } = useContext(ResultContext);

  return (
    <div className="container">
      Results page
      <br />
      <p>Food: {result.response.result_food_total}</p>
      <p>Services: {result.response.result_services_total}</p>
      <p>Housing: {result.response.result_housing_total}</p>
      <p>Transport: {result.response.result_transport_total}</p>
      <p>Goods: {result.response.result_goods_total}</p>
      <p>Total: {result.response.result_grand_total}</p>
      <button className="btn btn-primary"> Save Results</button>
    </div>
  );
};

export default Result;
