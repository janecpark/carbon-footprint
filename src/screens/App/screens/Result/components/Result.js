import React, { useContext } from 'react';
import { useHistory, Link } from 'react-router-dom';
import ResultContext from '../../../ResultContext';
import CarbonApi from '../../../CarbonApi';
import UserContext from '../../../UserContext';
import ActionContext from '../../../ActionContext';
import '../../../../../../node_modules/react-vis/dist/style.css';
import { XYPlot, XAxis, YAxis, VerticalBarSeries } from 'react-vis';

/** Renders result and chart to show the carbon footprint calculations
 * Option to save results and view ways to reduce emissions
 */

const Result = () => {
  const history = useHistory();
  const { result } = useContext(ResultContext);
  const { curUser } = useContext(UserContext);
  const { action } = useContext(ActionContext);
  let values = Object.values(result);
  let total = values.pop();
  let average = [
    { x: 'Global', y: 4 },
    { x: 'US', y: 16 },
  ];
  average.unshift({ x: 'Your Footprint', y: total });
  async function handleSubmit() {
    const res = await CarbonApi.sendResult(result, curUser);
    await CarbonApi.actionResult(action, res.result.id);
    history.push('/user-result');
  }

  const result_display_logged_in = (
    <div className="figures-display">
      <div className="row">
        <div className="col-md">
          <div className="figures">
            <div className="figures-total">{result.total}</div>
            <div className="figures-text text-right">
              Your Carbon Footprint (CO2 emissions/year tons)
            </div>
          </div>
          <hr />
          <div className="figures">
            <div className="figures-num">{result.food}</div>
            <div className="figures-text">Food</div>
          </div>
          <div className="figures">
            <div className="figures-num">{result.housing}</div>
            <div className="figures-text">Housing</div>
          </div>
          <div className="figures">
            <div className="figures-num">{result.transport}</div>
            <div className="figures-text">Transport</div>
          </div>
        </div>
        <div className="col-md graph">
          <p className='text-center'>Compare Average</p>
          <XYPlot height={300} width={300} xType="ordinal" stackBy="y">
            <VerticalBarSeries data={average} color="#CADFB6" />
            <XAxis />
            <YAxis />
          </XYPlot>
        </div>
      </div>
      <div className="d-flex justify-content-center figures-btn">
        <button onClick={handleSubmit} className="btn btn-primary result-btn">
          Save Results
        </button>
        <Link to="/take-action" className="btn btn-warning result-btn ml-2">
          Take Action
        </Link>
      </div>
    </div>
  );
  const result_display_logged_out = (
    <div className="figures-display">
      <div className="row">
        <div className="col-md">
          <div className="figures">
            <div className="figures-total">{result.total}</div>
            <div className="figures-text text-right">
              Your Carbon Footprint (CO2 emissions/year tons)
            </div>
          </div>
          <hr />
          <div className="figures">
            <div className="figures-num">{result.food}</div>
            <div className="figures-text">Food</div>
          </div>
          <div className="figures">
            <div className="figures-num">{result.housing}</div>
            <div className="figures-text">Housing</div>
          </div>
          <div className="figures">
            <div className="figures-num">{result.transport}</div>
            <div className="figures-text">Transport</div>
          </div>
        </div>
        <div className="col-md graph">
          <p className='text-center'>Compare Average</p>
          <XYPlot height={300} width={300} xType="ordinal" stackBy="y">
            <VerticalBarSeries data={average} color="#CADFB6" />
            <XAxis />
            <YAxis />
          </XYPlot>
        </div>
      </div>
      <div className="d-flex justify-content-center figures-btn">
        <Link to="/take-action" className="btn btn-warning ml-2">
          Take Action
        </Link>
      </div>
    </div>
  );

  const error_display = (
    <div className="error_display text-center">
      <h4>{result.errors}</h4>
      <Link to="/household1" className="btn btn-warning err-btn">
        Start Over
      </Link>
    </div>
  );

  return (
    <div className="Results">
      <h2 className="text-center mt-1">Results page</h2>
      <div className="Results-display">
        <br />
        {result.errors || Object.entries(result).length === 0 ? (
          <>
            <br />
            <p className="text-center">No results</p>
            {error_display}
          </>
        ) : (
          <>{curUser ? result_display_logged_in : result_display_logged_out}</>
        )}
      </div>
    </div>
  );
};

export default Result;
