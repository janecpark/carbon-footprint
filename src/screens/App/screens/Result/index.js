import Result from './components/Result';

export default Result;