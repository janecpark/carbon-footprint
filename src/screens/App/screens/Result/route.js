import { routeConstants } from '../../../../shared/constants';
import Result from "./";

export default {
    path: routeConstants.RESULT.route,
    component: Result
};