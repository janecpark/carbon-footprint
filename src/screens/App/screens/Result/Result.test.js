import React from 'react';
import { render } from '@testing-library/react';
import { MemoryRouter } from 'react-router';
import Result from './components/Result';
import {
  UserProvider,
  FormProvider,
  ResultProvider,
  ActionProvider,
} from '../testUtils';

it('renders without crashing', function () {
  render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <Result />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
});

it('matches snapshot', function () {
  const { asFragment } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <Result />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});


test('renders text on page', () => {
  const { getByText } = render(
    <MemoryRouter>
      <UserProvider>
        <FormProvider>
          <ResultProvider>
            <ActionProvider>
              <Result />
            </ActionProvider>
          </ResultProvider>
        </FormProvider>
      </UserProvider>
    </MemoryRouter>
  );
  const element = getByText(/Results Page/i);
  expect(element).toBeInTheDocument();
});