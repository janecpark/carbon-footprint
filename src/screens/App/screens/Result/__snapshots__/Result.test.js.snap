// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="Results"
  >
    <h2
      class="text-center mt-1"
    >
      Results page
    </h2>
    <div
      class="Results-display"
    >
      <br />
      <div
        class="figures-display"
      >
        <div
          class="row"
        >
          <div
            class="col-md"
          >
            <div
              class="figures"
            >
              <div
                class="figures-total"
              >
                12
              </div>
              <div
                class="figures-text text-right"
              >
                Your Carbon Footprint (CO2 emissions/year)
              </div>
            </div>
            <hr />
            <div
              class="figures"
            >
              <div
                class="figures-num"
              >
                3
              </div>
              <div
                class="figures-text"
              >
                Food
              </div>
            </div>
            <div
              class="figures"
            >
              <div
                class="figures-num"
              >
                4
              </div>
              <div
                class="figures-text"
              >
                Housing
              </div>
            </div>
            <div
              class="figures"
            >
              <div
                class="figures-num"
              >
                5
              </div>
              <div
                class="figures-text"
              >
                Transport
              </div>
            </div>
          </div>
          <div
            class="col-md graph"
          >
            <div
              class="rv-xy-plot "
              style="width: 300px; height: 300px;"
            >
              <svg
                class="rv-xy-plot__inner"
                height="300"
                width="300"
              >
                <g
                  class="rv-xy-plot__series rv-xy-plot__series--bar "
                  transform="translate(40,10)"
                >
                  <rect
                    height="150"
                    style="opacity: 1; stroke: #CADFB6; fill: #CADFB6;"
                    width="70.83333333333333"
                    x="6.250000000000007"
                    y="100"
                  />
                  <rect
                    height="200"
                    style="opacity: 1; stroke: #CADFB6; fill: #CADFB6;"
                    width="70.83333333333333"
                    x="89.58333333333334"
                    y="49.999999999999986"
                  />
                  <rect
                    height="250"
                    style="opacity: 1; stroke: #CADFB6; fill: #CADFB6;"
                    width="70.83333333333333"
                    x="172.91666666666666"
                    y="0"
                  />
                </g>
                <g
                  class="rv-xy-plot__axis rv-xy-plot__axis--horizontal "
                  transform="translate(40,260)"
                >
                  <line
                    class="rv-xy-plot__axis__line"
                    x1="0"
                    x2="250"
                    y1="0"
                    y2="0"
                  />
                  <g
                    class="rv-xy-plot__axis__ticks"
                    transform="translate(0, 0)"
                  >
                    <g
                      class="rv-xy-plot__axis__tick"
                      transform="translate(41.66666666666667, 0)"
                    >
                      <line
                        class="rv-xy-plot__axis__tick__line"
                        x1="0"
                        x2="0"
                        y1="-6"
                        y2="6"
                      />
                      <text
                        class="rv-xy-plot__axis__tick__text"
                        dy="0.72em"
                        text-anchor="middle"
                        transform="translate(0, 14)"
                      >
                        food
                      </text>
                    </g>
                    <g
                      class="rv-xy-plot__axis__tick"
                      transform="translate(125, 0)"
                    >
                      <line
                        class="rv-xy-plot__axis__tick__line"
                        x1="0"
                        x2="0"
                        y1="-6"
                        y2="6"
                      />
                      <text
                        class="rv-xy-plot__axis__tick__text"
                        dy="0.72em"
                        text-anchor="middle"
                        transform="translate(0, 14)"
                      >
                        housing
                      </text>
                    </g>
                    <g
                      class="rv-xy-plot__axis__tick"
                      transform="translate(208.33333333333331, 0)"
                    >
                      <line
                        class="rv-xy-plot__axis__tick__line"
                        x1="0"
                        x2="0"
                        y1="-6"
                        y2="6"
                      />
                      <text
                        class="rv-xy-plot__axis__tick__text"
                        dy="0.72em"
                        text-anchor="middle"
                        transform="translate(0, 14)"
                      >
                        transport
                      </text>
                    </g>
                  </g>
                </g>
                <g
                  class="rv-xy-plot__axis rv-xy-plot__axis--vertical "
                  transform="translate(0,10)"
                >
                  <line
                    class="rv-xy-plot__axis__line"
                    x1="40"
                    x2="40"
                    y1="0"
                    y2="250"
                  />
                  <g
                    class="rv-xy-plot__axis__ticks"
                    transform="translate(40, 0)"
                  >
                    <g
                      class="rv-xy-plot__axis__tick"
                      transform="translate(0, 250)"
                    >
                      <line
                        class="rv-xy-plot__axis__tick__line"
                        x1="6"
                        x2="-6"
                        y1="0"
                        y2="0"
                      />
                      <text
                        class="rv-xy-plot__axis__tick__text"
                        dy="0.32em"
                        text-anchor="end"
                        transform="translate(-14, 0)"
                      >
                        0
                      </text>
                    </g>
                    <g
                      class="rv-xy-plot__axis__tick"
                      transform="translate(0, 200)"
                    >
                      <line
                        class="rv-xy-plot__axis__tick__line"
                        x1="6"
                        x2="-6"
                        y1="0"
                        y2="0"
                      />
                      <text
                        class="rv-xy-plot__axis__tick__text"
                        dy="0.32em"
                        text-anchor="end"
                        transform="translate(-14, 0)"
                      >
                        1
                      </text>
                    </g>
                    <g
                      class="rv-xy-plot__axis__tick"
                      transform="translate(0, 150)"
                    >
                      <line
                        class="rv-xy-plot__axis__tick__line"
                        x1="6"
                        x2="-6"
                        y1="0"
                        y2="0"
                      />
                      <text
                        class="rv-xy-plot__axis__tick__text"
                        dy="0.32em"
                        text-anchor="end"
                        transform="translate(-14, 0)"
                      >
                        2
                      </text>
                    </g>
                    <g
                      class="rv-xy-plot__axis__tick"
                      transform="translate(0, 100)"
                    >
                      <line
                        class="rv-xy-plot__axis__tick__line"
                        x1="6"
                        x2="-6"
                        y1="0"
                        y2="0"
                      />
                      <text
                        class="rv-xy-plot__axis__tick__text"
                        dy="0.32em"
                        text-anchor="end"
                        transform="translate(-14, 0)"
                      >
                        3
                      </text>
                    </g>
                    <g
                      class="rv-xy-plot__axis__tick"
                      transform="translate(0, 49.999999999999986)"
                    >
                      <line
                        class="rv-xy-plot__axis__tick__line"
                        x1="6"
                        x2="-6"
                        y1="0"
                        y2="0"
                      />
                      <text
                        class="rv-xy-plot__axis__tick__text"
                        dy="0.32em"
                        text-anchor="end"
                        transform="translate(-14, 0)"
                      >
                        4
                      </text>
                    </g>
                    <g
                      class="rv-xy-plot__axis__tick"
                      transform="translate(0, 0)"
                    >
                      <line
                        class="rv-xy-plot__axis__tick__line"
                        x1="6"
                        x2="-6"
                        y1="0"
                        y2="0"
                      />
                      <text
                        class="rv-xy-plot__axis__tick__text"
                        dy="0.32em"
                        text-anchor="end"
                        transform="translate(-14, 0)"
                      >
                        5
                      </text>
                    </g>
                  </g>
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="d-flex justify-content-center figures-btn"
        >
          <button
            class="btn btn-primary result-btn"
          >
            Save Results
          </button>
          <a
            class="btn btn-warning result-btn ml-2"
            href="/take-action"
          >
            Take Action
          </a>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
